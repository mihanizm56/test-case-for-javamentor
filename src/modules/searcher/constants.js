const FRONTEND = 'FRONTEND'
const BACKEND = 'BACKEND'
const MOBILE = 'MOBILE'

export const listOptions = [{ value: FRONTEND }, { value: BACKEND }, { value: MOBILE }]
