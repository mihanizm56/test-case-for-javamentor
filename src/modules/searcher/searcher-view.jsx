import React, { memo } from 'react'
import MenuItem from '@material-ui/core/MenuItem'
import { Button, Select, LangList } from '../../components'

export const SearcherView = memo(
    ({ handleSearchClick, handleSearchChange, listData, options, selectedValue }) => (
        <div className='searcher-wrapper'>
            <div className='searcher-container'>
                <div className='searcher__select'>
                    <Select handleChange={handleSearchChange} selectedValue={selectedValue}>
                        {options.map((option, index) => (
                            <MenuItem className='searcher_option' value={option.value} key={index}>
                                {option.value}
                            </MenuItem>
                        ))}
                    </Select>
                </div>
                <Button handleClick={handleSearchClick} />
            </div>
            <div className='searcher-list'>
                <LangList listData={listData} />
            </div>
        </div>
    )
)
