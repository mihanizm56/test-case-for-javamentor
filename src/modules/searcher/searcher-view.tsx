import React, { memo,FunctionComponent } from 'react'
import MenuItem from '@material-ui/core/MenuItem'
import { Button, Select, LangList } from '../../components'
import {LangCardType} from '../../components/molecules'
import {OptionsType, SearcherViewProps} from './types'

export const SearcherView = memo(
    ({ handleSearchClick, handleSearchChange, langData, options, selectedValue } :SearcherViewProps) => (
        <div className='searcher-wrapper'>
            <div className='searcher-container'>
                <div className='searcher__select'>
                    <Select handleChange={handleSearchChange} selectedValue={selectedValue}>
                         {options.map((option:OptionsType, index:number):React.ReactNode => ( 
                            <MenuItem className='searcher_option' value={option.value} key={index}>
                                 {option.value}
                            </MenuItem>
                       ))}
                    </Select>
                </div>
                <Button handleClick={handleSearchClick} />
            </div>
            <LangList langData={langData} />
        </div>
    )
)
