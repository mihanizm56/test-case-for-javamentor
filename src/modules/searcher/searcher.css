.searcher-wrapper {
    align-self: center;
    width: 400px;
    overflow: hidden;
    background-color: #fff;
    padding: 10px 0;
}

.searcher-container {
    display: flex;
    justify-content: space-around;
    padding: 10px;
    width: 280px;
}

.searcher__select {
    width: 135px;
    height: 100%;
}

.searcher_option {
    width: 50px;
}