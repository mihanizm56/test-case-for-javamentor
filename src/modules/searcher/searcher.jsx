import React, { memo } from "react";
import { Button, Select, LangList } from "../../components";

export const Searcher = () => (
  <SearcherContainer>
    {({ handleSearchClick, handleSearchChange, listData }) => (
      <SearcherView
        handleSearchClick={handleSearchClick}
        handleSearchChange={handleSearchChange}
        listData={listData}
      />
    )}
  </SearcherContainer>
);
