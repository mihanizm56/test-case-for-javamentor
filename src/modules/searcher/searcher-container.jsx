import React, { Component } from "react";
import { connect } from "react-redux";
import { fetchLangData, getLangData } from "../../redux/modules/lang-data";
import { listOptions as options } from "./constants";

class WrappedContainer extends Component {
  state = {
    selectedField: null
  };

  handleSearchClick = () => {
    console.log("check handleSearchClick, should start ");
  };

  handleSearchChange = ({ target: { value } }) => {
    console.log("check handleSearchChange", value);
  };

  render = () => {
    const { listData, children } = this.props;

    return children({
      handleSearchClick: this.handleSearchClick,
      handleSearchChange: this.handleSearchChange,
      listData,
      options
    });
  };
}

export const SearcherContainer = connect(
  state => ({ getLangData: getLangData(state) }),
  { fetchLangData }
)(WrappedContainer);
