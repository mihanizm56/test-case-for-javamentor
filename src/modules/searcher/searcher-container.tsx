import React from "react";
import { connect } from "react-redux";
import { fetchLangData, getLangData } from "../../redux/modules/lang-data";
import { listOptions as options } from "./constants";
import { SearcherViewProps, handleSearchChangeParams } from "./types";
import { LangCardType } from "../../components/molecules";

type SearcherContainerPropsType = {
  fetchLangData: (value: string) => void;
  langData: Array<LangCardType>;
  children: (object: SearcherViewProps) => Array<React.ReactNode>;
};

type SearcherContainerStateType = {
  selectedValue: string;
};
class WrappedContainer extends React.Component<
  SearcherContainerPropsType,
  SearcherContainerStateType
> {
  state = {
    selectedValue: ""
  };

  handleSearchClick = () => this.props.fetchLangData(this.state.selectedValue);

  handleSearchChange = ({ target: { value } }: handleSearchChangeParams) =>
    this.setState({ selectedValue: value });

  render = () => {
    const { selectedValue } = this.state;
    const { langData, children } = this.props;
    const { handleSearchClick, handleSearchChange } = this;

    return children({
      handleSearchClick,
      handleSearchChange,
      selectedValue,
      langData,
      options
    });
  };
}

export const SearcherContainer: any = connect(
  getLangData,
  { fetchLangData }
)(WrappedContainer);
