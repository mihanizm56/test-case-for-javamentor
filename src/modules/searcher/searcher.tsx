import React, { memo } from 'react'
import { SearcherContainer } from './searcher-container'
import { SearcherView } from './searcher-view'
import {LangCardType} from '../../components/molecules'
import {OptionsType, handleSearchChangeParams} from './types'
import './searcher.css'



interface SearcherContainerRenderProps {
    handleSearchClick:()=>void, 
    handleSearchChange:(params:handleSearchChangeParams)=>void, 
    langData:Array<LangCardType>, 
    options:Array<OptionsType>, 
    selectedValue:string
}

export const Searcher = memo(() => (
    <SearcherContainer>
        {({ handleSearchClick, handleSearchChange, langData, options, selectedValue }: SearcherContainerRenderProps) => (
            <SearcherView
                handleSearchClick={handleSearchClick}
                handleSearchChange={handleSearchChange}
                langData={langData}
                options={options}
                selectedValue={selectedValue}
            />
        )}
    </SearcherContainer>
))
