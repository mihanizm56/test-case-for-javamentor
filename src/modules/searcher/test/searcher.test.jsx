import React from "react";
import { shallow } from "enzyme";
import { Searcher } from "../searcher";
import { SearcherContainer } from "../searcher-container";

it("the whole test Searcher by snapshot ", () => {
  const wrapper = shallow(<Searcher />)
    .find(SearcherContainer)
    .renderProp("children")({
    handleSearchClick: jest.fn(),
    handleSearchChange: jest.fn(),
    langData: [],
    options: {},
    selectedValue: "test"
  });

  expect(wrapper).toMatchSnapshot();
});
