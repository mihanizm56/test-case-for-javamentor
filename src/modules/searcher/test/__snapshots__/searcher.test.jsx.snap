// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test 1`] = `
<Memo()
  handleSearchChange={[MockFunction]}
  handleSearchClick={[MockFunction]}
  langData={Array []}
  options={Object {}}
  selectedValue="test"
/>
`;

exports[`the whole test Searcher by snapshot  1`] = `
<Memo()
  handleSearchChange={[MockFunction]}
  handleSearchClick={[MockFunction]}
  langData={Array []}
  options={Object {}}
  selectedValue="test"
/>
`;
